index.html

```diff
@@ -1,26 +1,125 @@
+<!-- working on send/add messages, see all messages and delete messages and give error
+ error message if not both name and message is filled in-->
+
 <!DOCTYPE html>
 <html lang="en">
   <head>
     <title>Welcome to CYF Chat</title>
   </head>
+
   <body>
-    <h1>
-      CYF Chat
-    </h1>
-    <h2>
-      Send a message
-    </h2>
+    <h1>CYF Chat</h1>
+    <h2>Send a message</h2>
     <form action="/messages" method="post">
       <p>
         Name: <input type="text" name="from" placeholder="Your Name" /> <br />
         Message: <input type="text" name="text" placeholder="The message..." />
         <br />
       </p>
-      <button type="submit">
-        Send
-      </button>
+      <button type="submit">Send</button>
     </form>
-
+    <br />
     <a href="/messages">See all messages</a>
+
+    <form>
+      <p>
+        Delete Message:
+        <input
+          type="number"
+          id="deleteId"
+          name="id"
+          placeholder="Delete message by ID..."
+        />
+      </p>
+      <button type="button" onclick="deleteMessage()">Submit</button>
+    </form>
+
+    <script>
+      function deleteMessage() {
+        const idToDelete = document.getElementById("deleteId").value;
+        if (!idToDelete || isNaN(idToDelete)) {
+          alert("Please enter a valid ID to delete.");
+          return;
+        }
+
+        const url = `/messages/delete/${idToDelete}`;
+        fetch(url, {
+          method: "DELETE",
+        })
+          .then((response) => {
```

Comment:
Also, as a suggestion and going forward with JavaScript, is to get acquainted with ES6+ features and apply them to your JS solutions. One of the ES6+ features (which generally means modern JavaScript) is the use of `async/await ` functions. There's this amazing instructor Colt Steele out there on Youtube, and I highly suggest you watch this tutorial on how to use it, here's the [link](https://www.youtube.com/watch?v=krAYA4rvbdA). He has really comprehensive and easy-to-follow tutorials on his Youtube channel. 10/10 recommended 🥹

The idea of using async/await is to make your codebase shorter and thus easier to understand. 😊